### PR TITLE
Feature/8689hx750/create-founder  - Feature/8689hx744/list-founder  - Feature/8689hx76a/edit-founder

### PR DESCRIPTION
**Pull request by CristhianRF24**

### CODE REVIEW

File: src/app/fundadores/create/page.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+  'use client'
```
Comment (richi0085) [inline src/app/fundadores/create/page.tsx]: Why is the route called 'Founders,' and then we have 'Create'? Please ensure that even the route names are in English.

so :
founders/create

File: src/app/fundadores/page.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  const handleDelete = () => {
+    fetchDataCallback();
+  };
+
+  return (
+    <>
+      <Title text="Lista De Fundadores"  href="/fundadores/create"/>
+      <SearchCard onSearchClick={handleSearchClick} ></SearchCard>
+            <TableSection found={data} searchTerm={searchTerm} handleDelete={handleDelete} itemType={"founders"}></TableSection>
```
Comment (richi0085) [inline src/app/fundadores/page.tsx]: too large please do:
	<TableSection
			found="{data}"
			searchTerm="{searchTerm}"
			handleDelete="{handleDelete}"
			itemType={"founders"}
		></TableSection>
	</div>

File: src/app/fundadores/page.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+            {!searchTerm && (<PaginationSection currentPage={currentPage} totalPages={Math.ceil(totalItems / itemsPerPage)} onPageChange={handlePageChange}></PaginationSection>)}
```
Comment (richi0085) [inline src/app/fundadores/page.tsx]: to large, please do:
	<PaginationSection
			currentPage="{currentPage}"
			totalPages={Math.ceil(totalItems / itemsPerPage)}
			onPageChange="{handlePageChange}"
		></PaginationSection>

File: src/components/delete-button.tsx [hunk 4]
```diff
@@ -1,10 +1,65 @@
+import { useState } from "react";
 import { Button } from "./ui/button";
 import { TrashIcon } from '@radix-ui/react-icons';
+import { get, ref, remove, set, update } from "firebase/database";
+import { database } from "@/app/firebaseConfig";
+import { AlertDialog, AlertDialogAction, AlertDialogCancel, AlertDialogContent, AlertDialogDescription, AlertDialogHeader, AlertDialogTitle, AlertDialogTrigger } from "./ui/alert-dialog";
```
Comment (richi0085) [inline src/components/delete-button.tsx]: too large please do:
import {
	AlertDialog,
	AlertDialogAction,
	AlertDialogCancel,
	AlertDialogContent,
	AlertDialogDescription,
	AlertDialogHeader,
	AlertDialogTitle,
	AlertDialogTrigger,
} from './ui/alert-dialog';

File: src/components/side-menu.tsx [hunk 5]
```diff
@@ -1,89 +1,191 @@
...
+                  width={125}
+                  height={100}
+                  alt="logo"
+                  className="mb-3 mt-7 cursor-pointer"
+                  onClick={handleImageClick}
+                />
+              </NavigationMenuItem>
+              <Accordion type="single" collapsible>
+                <NavigationMenuItem
+                  className={`w-full py-1 p-7 rounded-md transition-colors hover:bg-[#F2A594] ${getActiveClass(
```
Comment (richi0085) [inline src/components/side-menu.tsx]: refactor this to this:
className = {cn('w-full py-1 p-7 rounded-md transition-colors hover:bg-[#F2A594] ', getActiveClass(
                    MENU_ITEMS.STARTUPS.path
                  ))}

And fix this in all places where you have a dynamic class please

File: src/components/edit-button.tsx [hunk 6]
```diff
@@ -1,10 +1,27 @@
...
+  itemId: string;
+  itemType: string;
+}
+
+export default function EditButton({ itemId, itemType }: EditButtonProps) {
+  
+const router = useRouter();
+
+const handleClick =() => {
+  router.push(`/fundadores/edit/${itemId}`);
```
Comment (BelenA25) [inline src/components/edit-button.tsx]: We could make this more reusable by doing something like this: router.push(`/${itemType}/edit/${itemId}`);

**Review by MartinArispeRiveros — APPROVED** [review]

File: src/components/founder-form.tsx [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+    imagePreviewUrl,
+    handleFileChange,
+    uploadImage,
+    isUploading,
+    setImagePreviewUrl,
+  } = useImageUpload();
+
+  useEffect(() => {
+    if (founderId) {
+      getItemById(TYPE, founderId).then((founderData) => {
```
Comment (richi0085) [inline src/components/founder-form.tsx]: I prefer getFounderById is clear

File: src/components/search-card.tsx
Comment (richi0085) [inline src/components/search-card.tsx:1]: The component Name should be SearchCard.tsx

File: src/components/founder-form.tsx
Comment (richi0085) [inline src/components/founder-form.tsx:1]: The component name should be FounderForm.tsx

File: src/components/edit-button.tsx
Comment (richi0085) [inline src/components/edit-button.tsx:1]: The component name should be EditButton.tsx

File: src/components/delete-button.tsx
Comment (richi0085) [inline src/components/delete-button.tsx:1]: the file name should be DeleteButton.tsx

File: src/components/search-card.tsx
Comment (richi0085) [inline src/components/search-card.tsx:1]: the file name should be SearchCard.tsx

File: src/components/table-row-data.tsx
Comment (richi0085) [inline src/components/table-row-data.tsx:1]: fill all please and read this!
https://rajithasanjayamal.medium.com/naming-conventions-best-practices-in-react-37624d020288

**Review by richi0085 — CHANGES_REQUESTED** [review]

File: src/app/fundadores/create/page.tsx (hunk 1 above)
Comment (CristhianRF24) [reply src/app/fundadores/create/page.tsx]: I already corrected this in the last commitment I made

**Review by richi0085 — APPROVED** [review]

**Pull request merged**